design/proposals/spike-cli-dockerless.md

```diff
@@ -0,0 +1,128 @@
+# Spike - Containerless CLI
+
+## Table of Contents
+
+[//]: # "You can use this tool to generate a TOC - https://ecotrust-canada.github.io/markdown-toc/"
+
+  * [Useful links](#useful-links)
+  * [Background](#background)
+      - [Requirements](#requirements)
+  * [Research](#research)
+      - [Estimation](#estimation)
+      - [Takeways and recommendation](#takeways-and-recommendation)
+  * [Documentation](#documentation)
+  * [Open questions](#open-questions)
+  * [Implementation plan](#implementation-plan)
+
+## Useful links
+
+[//]: # "Add links that may be useful for the reader"
+
+| **Term**   | **Description**                                |
+| ---------- | ---------------------------------------------- |
+| Epic       | https://github.com/cyberark/conjur/issues/1892 |
+| Ruby CLI   | https://github.com/cyberark/conjur-cli         |
+| Golang CLI | https://github.com/cyberark/conjur-cli-go      |
+| Python CLI | https://github.com/cyberark/conjur-api-python3 |
+
+## Background
+
+[//]: # "Give relevant background for the designed feature. What is the motivation for this solution?"
+
+This document is one piece in the bigger piece of the Dockerless Conjur puzzle. We need to be able to deploy Conjur on a VM. This means we will need to figure out a way to deliver a containerless CLI to allow human-users to interface with Conjur.
+
+In this document we will discuss possible options for a containerless CLI and present a decision for which one we will pursue.
+
+#### Requirements
+
+The CLI design should answer the following requirements:
+
+*Core functionality*
+
+1. For each platform, we should support *init*, *login*, *authenticate*, *load policy*, and *fetch secrets* functionalities. These functionalities are must-haves but we should not be limited to providing just these.
+
+*Deployment*
+
+1. Provide a manual and automated way to deploy the CLI
+2. Require a maximum of 2 steps
+3. ***Recommended:*** cross-platform deployment (Linux, Windows) should all have the same flow
```

Comment:
That's a good point. I think we need to be very clear about what OS targets we want to support as not all solutions work equally well in them.

Reply:
@InbalZilberman do you have insights here regarding Rafi's questions on "Linux distro" and which macOS should be supported? 

> we want to support as not all solutions work equally well in them

& 

>  it might not work the same on both Debian and Fedora based systems

@sgnn7 @rafis3 Why not? isn't it just about compiling down the code to a binary that can run on each OS?

Reply:
A compiled binary should be compatible in both. But the installers are different - deb packages in Debian and rpm packages in Fedora. Also, for our tests, we need to know exactly where we need it to work, so we test accordingly and provide guidelines and assurance to our users, that it was tested and that it works on the stated platforms.

Reply:
Aren't we just supporting REHL (so that means Fedora) so why would also be supporting Debian?

What would testing look like for the different packages? Would it just be to the extent of running the package or would it be also testing all the major flows? @rafis3 @eladkug

Reply:
The CLI needs to run in the user's endpoint machine. That would mostly likely be Windows or macOS. Not as likely, Linux in my opinion.